tls/s2n_cipher_suites.c

```diff
@@ -1209,60 +1311,39 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
     const struct s2n_security_policy *security_policy;
     POSIX_GUARD(s2n_connection_get_security_policy(conn, &security_policy));
 
-    /* s2n supports only server order */
+    int server_index_of_chacha20_cipher = -1;
+    POSIX_GUARD_RESULT(s2n_get_index_of_chacha20_in_cipher_preferences(security_policy->cipher_preferences, &server_index_of_chacha20_cipher));
```

Comment:
Can we gate both of these calls behind the `if (prioritize_chacha)`? Otherwise we're iterating when we don't actually need to.

Reply:
Sure. I was worried about nesting if statements and thought the tradeoff might be ok. I'll change it

Reply:
If you want to avoid nested ifs you can always break this logic out into a separate function and return early (which is probably a good idea either way IMO)

https://github.com/aws/s2n-tls/blob/main/docs/DEVELOPMENT-GUIDE.md#development-principles

> Write clear readable code with a light cognitive load
s2n-tls's code must be concise, easy to follow and legible to a proficient C programmer. Our code should be organized in a way that divides the implementation up into small units of work, with the entire context necessary at hand.